extensions/amp-wpm-player/0.1/amp-wpm-player.js

```diff
@@ -0,0 +1,557 @@
+   * @param {string} data optional data to send with the command
+   * @param {boolean} skipQueue if this parameter is present the message will
+   * send the command even when the frame is not read
+   */
+  sendCommand_(name, data, skipQueue = false) {
+    if (this.frameReady_ || skipQueue) {
+      this.contentWindow_.postMessage(data
+        ? `${this.header_}${name}@PAYLOAD@${data}`
+        : `${this.header_}${name}`, '*');
+    } else {
+      this.messageQueue_.push({name, data});
+    }
+  }
+
+  /**
+   * Method that sends messages that were saved in queue
+   * @private
+   */
+  runQueue_() {
+    while (this.messageQueue_.length) {
+      const command = this.messageQueue_.shift();
+
+      this.sendCommand_(command.name, command.data);
+    }
+  }
+
+  /**
+   * Method that adds a listener for messages from iframe
+   * @private
+   * @param {string} messageName Name of the message this callback listenes for
+   * @param {function(string)} callback
+   */
+  addMessageListener_(messageName, callback) {
+    this.messageListeners_.push(data => {
+      const message = data.split('@PAYLOAD@');
+      if (messageName === message[0]) {
+        callback(message[1]);
+      }
+    });
+  }
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {!Element} */
+    this.element_ = element;
+
+    /** @private {!Array<Object<string, string>>} */
+    this.messageQueue_ = [];
+
+    /** @private {bool} */
+    this.frameReady_ = false;
+
+    /** @private {string} */
+    this.playingState_ = PlayingStates.PAUSED;
+
+    /** @private {!Array<function>} */
+    this.messageListeners_ = [];
+
+    /** @private {?Object} */
+    this.attributes_;
+
+    /** @private {string} */
+    this.frameId_;
+
+    /** @private {string} */
+    this.frameUrl_;
+
+    /** @private {string} */
+    this.videoId_;
+
+    /** @private {string} */
+    this.header_;
+
+    /** @private {element} */
+    this.container_;
+
+    /** @private {string} */
+    this.placeholderUrl_;
+
+    /** @private {element} */
+    this.iframe_;
+
+    /** @private {!Object} */
+    this.contentWindow_;
+
+    /** @private {number} */
+    this.position_;
+
+    /** @private {Array<Array<number>>} */
+    this.playedRanges_;
+
+    /** @private {Object} */
+    this.metadata_;
+
+    /** @private {!Window} */
+    this.win;
+  }
+
+  /** @override */
+  buildCallback() {
+    this.win.addEventListener('message', e => {
+      if (typeof e.data === 'string' && e.data.startsWith(this.header_)) {
+        const message = e.data.replace(this.header_, '');
+
+        this.messageListeners_.forEach(listener => {
+          listener(message);
+        });
+      }
+    });
+
+    this.attributes_ = this.parseAttributes_();
+
+    this.frameId_ = this.attributes_.id || `${Math.random() * 10e17}`;
+    this.frameUrl_ = new URL('https://std.wpcdn.pl/wpjslib/AMP-270-init-iframe/playerComponentFrame.html');
```

Comment:
- If possible, don't construct a `URL` object here. You can use `addQueryParamsToUrl` from `src/url` to help with creating a plain string. 
- Hoist a constant for the URL prefix since it's reused and you're cramping on line length here.